Dockerfile

```diff
@@ -0,0 +1,49 @@
+FROM php:8.3-apache
+
+# Update and install necessary packages
+RUN apt-get update && \
+    apt-get install -y libpng-dev libjpeg-dev libfreetype6-dev libopencv-dev && \
+    docker-php-ext-install gd && \
+    apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# Configure Apache
+RUN echo "<VirtualHost *:80>\n\
+    DocumentRoot /var/www/html\n\
+    <Directory /var/www/html>\n\
+    Options Indexes FollowSymLinks\n\
+    AllowOverride All\n\
+    Require all granted\n\
+    </Directory>\n\
+    # Redirect from root to Introduction.html\n\
+    RewriteEngine On\n\
+    RewriteCond %{REQUEST_URI} ^/$\n\
+    RewriteRule ^/$ /Introduction.html [R=301,L]\n\
+    </VirtualHost>" > /etc/apache2/sites-available/000-default.conf
```

Comment:
Do we need to enable this site using a2ensite? We will also need to understand where the https will terminate and configure this server accordingly.

Reply:
Since we are modifying `000-default.conf`, we do not need to enable it using a2ensite explicitly as apache reads this configuration by default. If we were creating a new site configuration file, then we would need to use a2ensite.

Also, the options for the `/var/www/html` directory were being set twice, so I will remove one of them.

For hhtps in docker, we would need to obtain a self-signed SSL certificate and ensure that Apache is configured to use the certificate. I think it would be easier to handle this from the AWS side and have the container expose http only.